### PR TITLE
Implement feature reminders and reminder history

### PR DESCRIPTION
**Pull request by giulls**
Feature reminders now work. Features are all possible data types from settings.
Alert and reminder checking is now a fragment. 
Reminders can be archived for maintenance history's sake, and reactivated. Archived reminders' data can be viewed, including date triggered.

### CODE REVIEW

File: app/src/main/java/com/teamltt/carcare/activity/HomeActivity.java [hunk 1]
```diff
@@ -89,7 +95,11 @@ protected void onCreate(Bundle savedInstanceState) {
         // Now start the new service
         startService(btServiceIntent);
 
+        DbHelper helper = new DbHelper(HomeActivity.this);
```
Comment (asasine) [inline app/src/main/java/com/teamltt/carcare/activity/HomeActivity.java]: a DbHelper instance is created several times in this class. Perhaps it should be a member?

File: app/src/main/java/com/teamltt/carcare/activity/HomeActivity.java [hunk 2]
```diff
@@ -117,6 +126,15 @@ protected void onStop() {
     }
 
     @Override
+    protected void onResume() {
+        super.onResume();
+
+        DbHelper helper = new DbHelper(HomeActivity.this);
```
Comment (asasine) [inline app/src/main/java/com/teamltt/carcare/activity/HomeActivity.java]: Same

File: app/src/main/java/com/teamltt/carcare/activity/ReminderActivity.java [hunk 3]
```diff
@@ -101,40 +135,118 @@ protected void updateUi(final Cursor info) {
         Button button;
 
         while (info.moveToNext()) {
```
Comment (asasine) [inline app/src/main/java/com/teamltt/carcare/activity/ReminderActivity.java:137]: This `info` variable is of `Cursor` type which should not exist beyond the `DbHelper` class.

File: app/src/main/java/com/teamltt/carcare/activity/ReminderActivity.java [hunk 4]
```diff
@@ -101,40 +135,118 @@ protected void updateUi(final Cursor info) {
...
-                public void onClick(View view) {
-                    editReminder(view, reminderId);
+            boolean isArchived = info.getInt(info.getColumnIndexOrThrow(ReminderContract.ReminderEntry.COLUMN_NAME_ARCHIVED)) != 0;
+            if (isArchived == showArchive) {
+                //create layout to hold row-/
+                LinearLayout dynamicLayout = new LinearLayout(this);
+                //adds edit button to
+                button = new Button(this);
+                button.setId(id);
+                final int reminderId = info.getInt(info.getColumnIndexOrThrow(ReminderContract.ReminderEntry.COLUMN_NAME_ID));
```
Comment (asasine) [inline app/src/main/java/com/teamltt/carcare/activity/ReminderActivity.java]: row ids are type `long` not `int`

**Review by asasine — CHANGES_REQUESTED** [review]
Semantic issues with the `DbHelper`

File: app/src/main/java/com/teamltt/carcare/activity/ReminderActivity.java (hunk 3 above)
Comment (LiveSatomi) [reply app/src/main/java/com/teamltt/carcare/activity/ReminderActivity.java:137]: I think we should not worry about this due to time constraints.

**Review by asasine — APPROVED** [review]

**Pull request merged**